host-controller/src/main/java/org/jboss/as/host/controller/operations/InstallationReportHandler.java

```diff
@@ -0,0 +1,92 @@
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.as.host.controller.HostControllerEnvironment;
+import org.jboss.as.host.controller.descriptions.HostResolver;
+import org.jboss.as.server.operations.AbstractInstallationReporter;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+/**
+ * Handler to produce a summary of the current host installation.
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a> (c) 2015 Red Hat, inc.
+ */
+public class InstallationReportHandler extends AbstractInstallationReporter {
+    public static final SimpleOperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(
+            OPERATION_NAME, HostResolver.getResolver(HOST))
+            .setRuntimeOnly()
+            .setReadOnly()
+            .setPrivateEntry()
+            .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.READ_WHOLE_CONFIG)
+            .withFlags(OperationEntry.Flag.DOMAIN_PUSH_TO_SERVERS)
+            .setReplyType(ModelType.OBJECT)
+            .setReplyParameters(SUMMARY_DEFINITION)
+            .build();
+
+    public static InstallationReportHandler createOperation(final HostControllerEnvironment environment) {
+        return new InstallationReportHandler(environment);
+    }
+
+    private final HostControllerEnvironment environment;
+
+    private InstallationReportHandler(HostControllerEnvironment environment) {
+        this.environment = environment;
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        final ModelNode patchingInfo = new ModelNode();
+        PathAddress patchingAddress = PathAddress.pathAddress(
+                PathElement.pathElement(HOST, environment.getHostControllerName()),
+                PathElement.pathElement(CORE_SERVICE, "patching"));
+        OperationEntry opEntry = context.getRootResourceRegistration().getOperationEntry(patchingAddress, "show-history");
+        if(opEntry != null) {
+            context.addStep(patchingInfo, Util.createOperation("patch-info", patchingAddress),
+                opEntry.getOperationHandler(), OperationContext.Stage.RUNTIME);
+        }
+        final Path installationDir = environment.getHomeDir().toPath();
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                System.out.println("Here is what we have as infos " + patchingInfo);
+                ModelNode result = context.getResult();
+                result.get(SUMMARY_DEFINITION.getName()).set(createProductNode(context, environment, patchingInfo, installationDir));
```

Comment:
The .get(SUMMARY_DEFINITION.getName()) shouldn't be necessary. If it is we need to figure out how to fix that.